implementation/src/main/java/io/smallrye/faulttolerance/SynchronousCircuitBreaker.java

```diff
@@ -207,10 +230,11 @@ private void resetCounts() {
 
     private final AtomicInteger successCount;
 
-    private final AtomicInteger failureCount;
-
     private final AtomicInteger halfOpenAttempts;
 
+    // failure = true, success = false
+    private final LinkedList<Boolean> rollingWindow;
```

Comment:
Won't it misbehave with multiple threads? I think `ConcurrentLinkedQueue` should fit and be thread-safe.

Reply:
Well, it's only manipulated from within synchronized methods. But `ConcurrentLinkedDeque` might be a better fit.

Reply:
Doesn't the rolling window have a fixed size? Array-backed ring buffer should perform better than a linked list.

Reply:
It has and it should. But I don't think it makes sense to save few ms here.

Reply:
The difference probably isn't even in millis :-) I actually think that plain `ArrayList` outperforms a `LinkedList` in almost all cases, but don't really mind.